gitmanager.py

```diff
@@ -306,20 +307,56 @@ def remove_from_blacklist(cls, item, username, blacklist_type="", code_privilege
     @classmethod
     def merge_pull_request(cls, pr_id, comment=""):
         response = requests.get("https://api.github.com/repos/{}/pulls/{}".format(GlobalVars.bot_repo_slug, pr_id))
-        if not response:
+        files = requests.get("https://api.github.com/repos/{}/pulls/{}/files".format(GlobalVars.bot_repo_slug, pr_id))
+        if not response or not files:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
-        if pr_info["user"]["login"] != "SmokeDetector":
+        files_info = files.json()
+        if pr_info["user"]["login"] != "Daniil-SD":
             raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
         if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
             raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        string = pr_info["title"]
+        file = files_info[0]["filename"]
+        print(file)
         ref = pr_info['head']['ref']
+        string1 = regex.match(r".*?: \S+ (.*?)(?:\Z|\s*\(\?#)", string).group(1)
```

Comment:
This appears to strip off any regex comment. While we may want the general case to be comparison without comments, we currently are not doing that for validation. If we do that, then it should be implemented in `Blacklist.exists()`.